azurelinuxagent/common/protocol/restapi.py

```diff
@@ -238,6 +238,9 @@ def detect(self):
     def update_goal_state(self):
         raise NotImplementedError()
 
+    def try_update_goal_state(self):
```

Comment:
Small suggestion, why not retire the public `update_goal_state` method for `try_update_goal_state` one to ensure everyone only uses this public function.

Reply:
both of them will go away in favor of the get_goal_state function we've talked about. The goal state is still stored in the protocol client and we need to get it our of there. Will continue the refactoring once metadata is gone. Not only Metadata is a pain to update, but I do not want to take the risk of breaking it, since we do not automation for it.